src/internal/connector/exchange/api/contacts.go

```diff
@@ -55,6 +55,41 @@ func (c Contacts) CreateContactFolder(
 	return mdl, nil
 }
 
+// GetContactFolderByDisplayName fetches contact folder which has the displayName of
+// folderName. This is only done at root level.
+func (c Contacts) GetContactFolderByDisplayName(
+	ctx context.Context,
+	user, folderName string,
+) (models.ContactFolderable, error) {
+	filter := fmt.Sprintf("displayName eq '%s'", folderName)
+	options := &users.ItemContactFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemContactFoldersRequestBuilderGetQueryParameters{
+			Filter: &filter,
+		},
+	}
+
```

Comment:
Since you're adding `folder_name` to multiple error returns, I'd suggest adding it to the context here instead, to make sure that all subsequent errors incorporate it:
```suggestion
ctx = clues.Add(ctx, "folder_name", folderName)
```